mem/test_mprotect_fault.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-test
+ *
+ * mprotect syscall tests
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Hubert Badocha
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/mman.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "unity_fixture.h"
+
+
+TEST_GROUP(test_mprotect_fault);
+
+
+TEST_SETUP(test_mprotect_fault)
+{
+}
+
+
+TEST_TEAR_DOWN(test_mprotect_fault)
+{
+}
+
+
+TEST(test_mprotect_fault, unit)
+{
+	long pageSize = sysconf(_SC_PAGESIZE);
+	long totalSize = 4 * pageSize;
+	volatile unsigned char *area = mmap(NULL, totalSize, PROT_READ | PROT_WRITE, MAP_ANONYMOUS, -1, 0);
+	TEST_ASSERT_NOT_NULL(area);
+
+	TEST_ASSERT_EQUAL(0, mprotect((void *)area, totalSize, PROT_READ));
+
+	area[0] = 0x42;                   /* Should generate fault. */
+	TEST_ASSERT_EQUAL(0x42, area[0]); /* Here to make sure compiler doesn't optimize away write. */
```

Comment:
Hmm, not needed, volatile musn't be optimized-out, also if we managed to protect this memory it cannot assume value of 0x42
```suggestion
	TEST_ASSERT_NOT_EQUAL(0x42, area[0]);
```